arangod/Replication2/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+add_library(arango_replication2 STATIC
+  AgencyCollectionSpecification.cpp
+  AgencyMethods.cpp
+  Methods.cpp
+  MetricsHelper.cpp
+  Version.cpp
+  ${PROJECT_SOURCE_DIR}/arangod/RestHandler/RestLogHandler.cpp
+  ${PROJECT_SOURCE_DIR}/arangod/RestHandler/RestLogInternalHandler.cpp
+  ${PROJECT_SOURCE_DIR}/arangod/RestHandler/RestPrototypeStateHandler.cpp)
+
+target_link_libraries(arango_replication2
+  arango
+	arango_network
+  arangoserver
+  fmt
+  immer)
+
+target_include_directories(arango_replication2 PRIVATE
+  "${PROJECT_SOURCE_DIR}/arangod"
+  "${PROJECT_SOURCE_DIR}/${ENTERPRISE_INCLUDE_DIR}")
+
+target_compile_options(arango_replication2 PRIVATE
+	${CMAKE_DEPRECATE_OPTIONS})
+
+add_subdirectory(Exceptions)
+add_subdirectory(ReplicatedLog)
+add_subdirectory(ReplicatedState)
+add_subdirectory(StateMachines)
```

Comment:
There are .tpp files in the Streams subdirectory but they were not mentioned in any CMakeLists. I'm wondering whether this was intentional, possibly I should add them?

Reply:
Ok, I added the tpp files as well.